File: Ejercicio 14/Ejercicio 14.py
import random

# Listas enlazadas
class Nodo:
    def __init__(self, valor):
        self.valor = valor
        self.siguiente = None

def main():
    cabeza = None
    actual = None
    
    print("Generando 15 valores aleatorios")
    print(" ")
    
    # Crear 15 nodos con valores aleatorios
    for i in range(15):
        valor_aleatorio = 1 + random.randint(0, 99)  # 1-100
        nuevo_nodo = Nodo(valor_aleatorio)
        
        if cabeza is None:
            cabeza = nuevo_nodo
            actual = cabeza
        else:
            actual.siguiente = nuevo_nodo
            actual = nuevo_nodo
    
    # Mostrar la lista
    print("Lista Enlazada:")
    actual = cabeza
    contador = 1
    while actual is not None:
        print(f"Nodo {contador}: {actual.valor}")
        actual = actual.siguiente
        contador += 1

# Ejecutar el programa
if __name__ == "__main__":
    main()

File: Ejercicio 14/Ejercicio 14.js
//Listas enlazadas
class Nodo {
    constructor(valor) {
        this.valor = valor;
        this.siguiente = null;
    }
}

function main() {
    let cabeza = null;
    let actual = null;
    
    console.log("Generando 15 valores aleatorios");
    console.log(" ");
    
    // Crear 15 nodos con valores aleatorios
    for (let i = 0; i < 15; i++) {
        const valorAleatorio = 1 + Math.floor(Math.random() * 100); // 1-100
        const nuevoNodo = new Nodo(valorAleatorio);
        
        if (cabeza === null) {
            cabeza = nuevoNodo;
            actual = cabeza;
        } else {
            actual.siguiente = nuevoNodo;
            actual = nuevoNodo;
        }
    }
    
    // Mostrar la lista
    console.log("Lista Enlazada:");
    actual = cabeza;
    let contador = 1;
    while (actual !== null) {
        console.log(`Nodo ${contador}: ${actual.valor}`);
        actual = actual.siguiente;
        contador++;
    }
}
main();